clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1006,7 +1004,14 @@ public void assign(Collection<TopicPartition> partitions) {
         fetcher.sendFetches();
 
         long now = time.milliseconds();
-        client.poll(Math.min(coordinator.timeToNextPoll(now), timeout), now);
+        long pollTimeout = Math.min(coordinator.timeToNextPoll(now), timeout);
+
+        client.poll(pollTimeout, now, new PollCondition() {
+            @Override
+            public boolean pollNeeded() {
+                return !fetcher.hasCompletedFetches();
+            }
+        });
```

Comment:
Good question. We should be fine if we get some response from the fetch request since we save error checking for when we parse the completed fetches. However, that still leaves the possibility of a disconnect or connection timeout. In addition to checking for completed fetches, we probably need to check whether there are any fetches in flight that we need to block for.